src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -184,6 +184,26 @@ CHIP_ERROR ConfigurationManagerImpl::StoreTotalOperationalHours(uint32_t totalOp
     return WriteConfigValue(ESP32Config::kCounterKey_TotalOperationalHours, totalOperationalHours);
 }
 
+CHIP_ERROR ConfigurationManagerImpl::GetProductURL(char * buf, size_t bufSize)
+{
+    CHIP_ERROR err = ReadConfigValueStr(ESP32Config::kConfigKey_ProductURL, buf, bufSize, bufSize);
+    if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
```

Comment:
Would `CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND` be more accurate? That would convert a `not found` into a `not found` rather than unsupported.

Reply:
Changed error to `CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND`.